src/av2/evaluation/forecasting/eval.py

```diff
@@ -310,18 +330,92 @@ def convert_forecast_labels(labels: Dict[str, List[Dict[str, Any]]]) -> Sequence
+    """Remove all tracks that are beyond the max_dist.
+
+    Args:
+        forecasts: Dict[seq_id: List[frame]] Dictionary of tracks
+        max_range_m: maximum distance from ego-vehicle
+
+    Returns:
+        forecasts: Dict[seq_id: List[frame]] Dictionary of tracks.
+    """
+    for seq_id in forecasts.keys():
+        for timestamp in forecasts[seq_id].keys():
+            keep_forecasts = [
+                agent
+                for agent in forecasts[seq_id][timestamp]
+                if np.linalg.norm(agent["current_translation"] - agent["ego_translation"]) < max_range_m
+            ]
+            forecasts[seq_id][timestamp] = keep_forecasts
+
+    return forecasts
+
+
+def yaw_to_quaternion3d(yaw: float) -> np.ndarray:
+    """Convert a rotation angle in the xy plane (i.e. about the z axis) to a quaternion.
+
+    Args:
+        yaw: angle to rotate about the z-axis, representing an Euler angle, in radians
+    Returns:
+        array w/ quaternion coefficients (qw,qx,qy,qz) in scalar-first order, per Argoverse convention.
+    """
+    qx, qy, qz, qw = Rotation.from_euler(seq="z", angles=yaw, degrees=False).as_quat()
+    return np.array([qw, qx, qy, qz])
+
+
+def filter_drivable_area(forecasts: Sequences, dataset_dir: str) -> Sequences:
+    """Convert the unified label format to a format that is easier to work with for forecasting evaluation.
+
+    Args:
+        forecasts: Dict[seq_id: List[frame]] Dictionary of tracks
+        dataset_dir:str Dataset root directory
+    Returns:
+        forecasts: Dict[seq_id: List[frame]] Dictionary of tracks.
+    """
+    if dataset_dir is None:
+        return tracks
+
+    log_ids = list(forecasts.keys())
+    log_id_to_avm, log_id_to_timestamped_poses = load_mapped_avm_and_egoposes(log_ids, Path(dataset_dir))
+
+    for log_id in log_ids:
+        avm = log_id_to_avm[log_id]
+
+        for timestamp in forecasts[log_id]:
+            city_SE3_ego = log_id_to_timestamped_poses[log_id][int(timestamp)]
+
+            translation, size, quat = [], [], []
+
+            for box in forecasts[log_id][timestamp]:
+                translation.append(box["current_translation"] - box["ego_translation"])
+                size.append(box["size"])
+                quat.append(yaw_to_quaternion3d(box["yaw"]))
+
+            score = np.ones((len(translation), 1))
+            boxes = np.concatenate([np.array(translation), np.array(size), np.array(quat), np.array(score)], axis=1)
+
+            is_evaluated = compute_objects_in_roi_mask(boxes, city_SE3_ego, avm)
+            forecasts[log_id][timestamp] = list(np.array(forecasts[log_id][timestamp])[is_evaluated])
+
+    return forecasts
+
+
 if __name__ == "__main__":
     argparser = argparse.ArgumentParser()
-    argparser.add_argument("--predictions", default="sample/constant_velocity.pkl")
+    argparser.add_argument("--predictions", default="sample/linear_forecast_predictions.pkl")
```

Comment:
Are these files we provide with the API? Or just a default naming scheme we assume?

Reply:
Artifact of testing. Removing default and updating with required=True